---
title: Derleyici Hatası CS1626
ms.date: 07/20/2015
f1_keywords:
- CS1626
helpviewer_keywords:
- CS1626
ms.assetid: 3ba03383-eb24-4fd8-bf40-8b0f7d6baf0d
ms.openlocfilehash: 3d24280006d77c807109723cd05d2836570fdc91
ms.sourcegitcommit: 3d5d33f384eeba41b2dff79d096f47ccc8d8f03d
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 05/04/2018
ms.locfileid: "33308213"
---
# <a name="compiler-error-cs1626"></a><span data-ttu-id="4e5dc-102">Derleyici Hatası CS1626</span><span class="sxs-lookup"><span data-stu-id="4e5dc-102">Compiler Error CS1626</span></span>
<span data-ttu-id="4e5dc-103">Bir değer deneyin blok catch yan tümcesi ile gövdesinde verim olamaz</span><span class="sxs-lookup"><span data-stu-id="4e5dc-103">Cannot yield a value in the body of a try block with a catch clause</span></span>  
  
 <span data-ttu-id="4e5dc-104">Try blok ile ilişkili bir catch yan tümcesi ise verim deyimi bir try bloğu içinde izin verilmiyor.</span><span class="sxs-lookup"><span data-stu-id="4e5dc-104">A yield statement is not allowed in a try block if there is a catch clause associated with the try block.</span></span> <span data-ttu-id="4e5dc-105">Bu hatayı önlemek için catch yan tümcesi dışında verim deyimi taşıyın.</span><span class="sxs-lookup"><span data-stu-id="4e5dc-105">To avoid this error, move the yield statement out of the catch clause.</span></span>  
  
 <span data-ttu-id="4e5dc-106">Aşağıdaki örnek CS1626 oluşturur:</span><span class="sxs-lookup"><span data-stu-id="4e5dc-106">The following sample generates CS1626:</span></span>  
  
```csharp  
// CS1626.cs  
using System.Collections;  
  
class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()  
   {  
      try  
      {  
         yield return this;  // CS1626  
      }  
      catch  
      {  
  
      }  
   }  
}  
  
public class CMain  
{  
   public static void Main() { }  
}  
```
